knack/parser.py

```diff
@@ -10,6 +10,22 @@
 from .util import CtxTypeError
 
 
+# List of keyword arguments supported in argparse
+# from https://github.com/python/cpython/blob/master/Lib/argparse.py#L748
+ARGPARSE_SUPPORTED_KWARGS = [
+    'option_strings',
+    'dest',
+    'nargs',
+    'const',
+    'default',
+    'type',
+    'choices',
+    'required',
+    'help',
+    'metavar',
+]
```

Comment:
In your 'KnackConversion' branch, [you also included 'action'](https://github.com/Azure/azure-cli/blob/KnackConversion/src/azure-cli-core/azure/cli/core/parser.py#L21) but I haven't included it in the PR to Knack as I didn't see it in argparse in https://github.com/python/cpython/blob/master/Lib/argparse.py#L748.

Reply:
Is this good or is there a scenario where 'action' should be added?

Reply:
Action is definitely supported by argparse and thus should be in the list.

Reply:
`action='store_true'`  is an example.